src/test/java/org/opensearch/plugin/insights/core/service/categorizor/SearchQueryCategorizerTests.java

```diff
@@ -54,12 +59,17 @@ public final class SearchQueryCategorizerTests extends OpenSearchTestCase {
 
     private SearchQueryCategorizer searchQueryCategorizer;
 
+    private Map<MetricType, Number> measurements;
+
     @Before
     public void setup() {
         metricsRegistry = mock(MetricsRegistry.class);
         when(metricsRegistry.createCounter(any(String.class), any(String.class), any(String.class))).thenAnswer(
             invocation -> mock(Counter.class)
         );
+        when(metricsRegistry.createHistogram(any(String.class), any(String.class), any(String.class))).thenAnswer(
+            invocation -> mock(Histogram.class)
+        );
```

Comment:
Since we are controlling the initialization of histogram object, can you use the `search.query.type.latency.histogram` string to get access to the mock object, instead of adding getters in the SearchQueryCounters?

Reply:
Makes sense, created a lookup map to get the histogram based on the name and deleted the getters.